pkg/apis/triggers/v1alpha1/interceptor_type_types.go

```diff
@@ -0,0 +1,103 @@
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// Check that EventListener may be validated and defaulted.
+var _ apis.Validatable = (*InterceptorType)(nil)
+var _ apis.Defaultable = (*InterceptorType)(nil)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genreconciler:krshapedlogic=false
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// InterceptorType describes a pluggable interceptor including configuration
+// such as the fields it accepts and its deployment address
+type InterceptorType struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec InterceptorTypeSpec `json:"spec"`
+	// +optional
+	Status InterceptorTypeStatus `json:"status"`
+}
+
+// InterceptorTypeSpec describes the Spec for an InterceptorType
+type InterceptorTypeSpec struct {
+	ClientConfig ClientConfig `json:"clientConfig"`
```

Comment:
Should we reuse [WebhookClientConfig](https://github.com/kubernetes/apiextensions-apiserver/blob/b343cbeb89f1b1f00514360d1e67397207b2e7be/pkg/apis/apiextensions/types.go#L107)? 

At the very least we can likely reuse the ServiceReference if we want to use the apis.URL for URL. The cert data is also useful for HTTPS.

Reply:
I think using a WebhookClientConfig is a good idea - maybe just reimplementing the type here if we are concerned about dependencies.

Reply:
yeah, I am basically reimplementing the type here....two differences:
1. using apis.URL instead of string for the URL type for better validation/json marshalling
2. we do not have a `caCert` field type (yet!) for HTTPS


Pulling in a new dependency for just the two structs  seems much so I decided to copy the structs. It also allows us to make changes (e.g. making the port a *int instead of int as suggested below)